src/stores/external-source.ts

```diff
@@ -0,0 +1,176 @@
+            externalSource.pkey.key === entry.external_source_name,
+        )?.created_at ?? '';
+      if (res[entry.username]) {
+        res[entry.username].push({
+          change_date,
+          derivation_group_name: entry.derivation_group,
+          key: entry.external_source_name,
+          source_type_name: entry.external_source_type,
+        });
+      } else {
+        res[entry.username] = [
+          {
+            change_date,
+            derivation_group_name: entry.derivation_group,
+            key: entry.external_source_name,
+            source_type_name: entry.external_source_type,
+          },
+        ];
+      }
+    }
+    return res;
+  },
+);
+
+/* Helper Functions. */
+export function resetExternalSourceStores(): void {
+  createExternalSourceError.set(null);
+  createExternalSourceTypeError.set(null);
+  createDerivationGroupError.set(null);
+  derivationGroupPlanLinkError.set(null);
+}
+
+function transformExternalSources(
+  externalSources:
+    | {
+        created_at: string;
+        derivation_group_name: string;
+        end_time: string;
+        key: string;
+        source_type_name: string;
+        start_time: string;
+        valid_at: string;
+      }[]
+    | null
+    | undefined,
+): ExternalSourceSlim[] {
+  const completeExternalSourceSlim: ExternalSourceSlim[] = [];
+  if (externalSources !== null && externalSources !== undefined) {
+    externalSources.forEach(externalSource => {
+      completeExternalSourceSlim.push({
+        created_at: externalSource.created_at,
+        end_time: externalSource.end_time,
+        pkey: {
+          derivation_group_name: externalSource.derivation_group_name,
+          key: externalSource.key,
+        },
+        source_type_name: externalSource.source_type_name,
+        start_time: externalSource.start_time,
+        valid_at: externalSource.valid_at,
+      });
+    });
+  }
+  return completeExternalSourceSlim;
+}
+
+function transformDerivationGroups(
+  derivationGroups:
+    | {
+        derived_total: number;
+        event_types: string[];
+        name: string;
+        source_type_name: string;
+        sources: string[];
+      }[]
+    | null
+    | undefined,
+): DerivationGroup[] {
+  const completeExternalSourceSlim: DerivationGroup[] = [];
+  if (derivationGroups !== null && derivationGroups !== undefined) {
+    derivationGroups.forEach(derivationGroup => {
+      completeExternalSourceSlim.push({
+        derived_event_total: derivationGroup.derived_total,
+        event_types: derivationGroup.event_types,
+        name: derivationGroup.name,
+        source_type_name: derivationGroup.source_type_name,
+        sources: new Map(
+          // comes from view schema that is hardcoded as "{source_key}, {derivation_group_name}, {event_count}""
+          derivationGroup.sources
+            .filter(source => source.charAt(0) !== ',' && source.length > 4)
```

Comment:
This filter is present to handle empty derivation groups. If a derivation group is empty, instead of an empty list the view `derivation_group_comp` produces a list with a singular element `", , "`, so this filter catches that.

Reply:
A source being ", ," is the default return from the database if the derivation group has no sources. So currently I believe the filter should exist to not do any work transforming the non-existent sources - with that said, should we instead try and change that default return to an empty array or null or something?